### ISSUE TITLE
MQ baseline inference warnings

### ISSUE THREAD

**Issue by JunweiLiang** [EGO4D/episodic-memory#29]
Hi,
I have followed the latest command to generate clip_annotations.json and training. During inferencing on the validation set, I encountered quite a few of these warnings:
```
Infer.py:145: RuntimeWarning: invalid value encountered in true_divide
  ovr = inter / (lengths[i] + lengths[order[1:]] - inter)
```
It seems the NMS failed on some samples. Is this the expected behavior? It did finish without errors. But the evaluation on validation is quite bad, only 0.05 Average-mAP (vs 0.24 in the paper?). 

My inference command:
```
$ python Infer.py --use_xGPN  --is_train false --dataset ego4d --feature_path /youtu_pedestrian_detection/junweil/ego4d/ego4d_data/v1/slowfast8x8_r101_k400/  --checkpoint_path ../baseline_slowfast_vsgn_yesxgpn  --output_path ../baseline_slowfast_vsgn_yesxgpn_output --clip_anno ../../../ego4d_data/v1/annotations/moments_official_clip_annotations.json --moment_classes moment_classes_idx.json --batch_size 2 --infer_datasplit validation
```

Evaluation command:
```
$ python Eval.py --dataset ego4d --output_path ../baseline_slowfast_vsgn_yesxgpn_output --out_prop_map true --eval_stage all  --clip_anno ../../../ego4d_data/v1/annotations/moments_official_clip_annotations.json --moment_classes moment_classes_idx.json --infer_datasplit validation
```

Evaluation outputs:
```
{'dataset': 'ego4d', 'is_train': 'true', 'out_prop_map': 'true', 'feature_path': '/mnt/sdb1/Datasets/Ego4d/action_feature_canonical', 'clip_anno': '../../../ego4d_data/v1/annotations/moments_official_clip_annotations.json', 'moment_classes': 'moment_classes_idx.json', 'checkpoint_path': 'checkpoint', 'output_path': '../baseline_slowfast_vsgn_yesxgpn_output', 'prop_path': 'proposals', 'prop_result_file': 'proposals_postNMS.json', 'detect_result_file': 'detections_postNMS.json', 'retrieval_result_file': 'retrieval_postNMS.json', 'detad_sensitivity_file': 'detad_sensitivity', 'batch_size': 32, 'train_lr': 5e-05, 'weight_decay': 0.0001, 'num_epoch': 30, 'step_size': 15, 'step_gamma': 0.1, 'focal_alpha': 0.01, 'nms_alpha_detect': 0.46, 'nms_alpha_prop': 0.75, 'nms_thr': 0.4, 'temporal_scale': 928, 'input_feat_dim': 2304, 'bb_hidden_dim': 256, 'decoder_num_classes': 111, 'num_levels': 5, 'num_head_layers': 4, 'nfeat_mode': 'feat_ctr', 'num_neigh': 12, 'edge_weight': 'false', 'agg_type': 'max', 'gcn_insert': 'par', 'iou_thr': [0.5, 0.5, 0.7], 'anchor_scale': [1, 10], 'base_stride': 1, 'stitch_gap': 30, 'short_ratio': 0.4, 'clip_win_size': 0.38, 'use_xGPN': False, 'use_VSS': False, 'num_props': 200, 'tIoU_thr': [0.1, 0.2, 0.3, 0.4, 0.5], 'eval_stage': 'all', 'infer_datasplit': 'validation'}
---------------------------------------------------------------------------------------------
2. Detection evaluation starts!
---------------------------------------------------------------------------------------------
a. Generate detections!
/youtu_pedestrian_detection/junweil/ego4d/mq_task/official_code/MQ/Evaluation/ego4d/generate_detection.py:14: RuntimeWarning: invalid value encountered in double_scalars
  return float(Aand) / Aor
/youtu_pedestrian_detection/junweil/ego4d/mq_task/official_code/MQ/Evaluation/ego4d/generate_detection.py:14: RuntimeWarning: invalid value encountered in double_scalars
  return float(Aand) / Aor
...
b. Evaluate the detection results!
{'take_photo_/_record_video_with_a_camera': 0, 'hang_clothes_in_closet_/_on_hangers': 1, 'browse_through_clothing_items_on_rack_/_shelf_/_hanger': 2, 'withdraw_money_from_atm_/_operate_atm': 3, 'stir_/_mix_ingredients_in_a_bowl_or_pan_(before_cooking)': 4, 'wash_hands': 5, 'clean_/_wipe_other_surface_or_object': 6, 'put_away_(or_take_out)_ingredients_in_storage': 7, 'throw_away_trash_/_put_trash_in_trash_can': 8, 'turn-on_/_light_the_stove_burner': 9, 'arrange_/_organize_items_in_fridge': 10, 'converse_/_interact_with_someone': 11, 'climb_up_/_down_a_ladder': 12, 'plaster_wall_/_surface': 13, 'paint_using_paint_brush_/_roller': 14, 'use_a_vacuum_cleaner_to_clean': 15, 'use_phone': 16, 'watch_television': 17, 'dismantle_other_item': 18, 'drill_into_wall_/_wood_/_floor_/_metal': 19, 'fix_other_item': 20, 'stir_/_mix_food_while_cooking': 21, 'knead_/_shape_/_roll-out_dough': 22, 'clean_/_wipe_kitchen_appliance': 23, 'arrange_/_organize_other_items': 24, 'cut_dough': 25, 'fix_wiring': 26, 'cut_other_item_using_tool': 27, 'read_a_book_/_magazine_/_shopping_list_etc.': 28, 'clean_/_wipe_a_table_or_kitchen_counter': 29, 'walk_down_stairs_/_walk_up_stairs': 30, 'place_items_in_shopping_cart': 31, 'browse_through_groceries_or_food_items_on_rack_/_shelf': 32, '"clean_/_repair_small_equipment_(mower,_trimmer_etc.)"': 33, 'use_hammer_/_nail-gun_to_fix_nail': 34, 'measure_wooden_item_using_tape_/_ruler': 35, 'mark_item_with_pencil_/_pen_/_marker': 36, 'use_a_laptop_/_computer': 37, 'fry_other_food_item': 38, 'put_away_(or_take_out)_food_items_in_the_fridge': 39, 'count_money_before_paying': 40, 'pack_food_items_/_groceries_into_bags_/_boxes': 41, 'pay_at_billing_counter': 42, 'browse_through_accessories_on_rack_/_shelf': 43, 'cut_thread_/_paper_/_cardboard_using_scissors_/_knife_/_cutter': 44, 'dig_or_till_the_soil_with_a_hoe_or_other_tool': 45, '"level_ground_/_soil_(eg._using_rake,_shovel,_etc)"': 46, 'pack_soil_into_the_ground_or_a_pot_/_container': 47, 'plant_seeds_/_plants_/_flowers_into_ground': 48, 'enter_a_supermarket_/_shop': 49, 'browse_through_other_items_on_rack_/_shelf': 50, 'exit_a_supermarket_/_shop': 51, '"put_on_safety_equipment_(e.g._gloves,_helmet,_safety_goggles)"': 52, 'stand_in_the_queue_/_line_at_a_shop_/_supermarket': 53, 'weigh_food_/_ingredient_using_a_weighing_scale': 54, 'arrange_/_organize_clothes_in_closet/dresser': 55, 'fold_clothes_/_sheets': 56, 'fry_dough': 57, 'remove_food_from_the_oven': 58, 'water_soil_/_plants_/_crops': 59, 'play_board_game_or_card_game': 60, 'clean_/_sweep_floor_with_broom': 61, 'eat_a_snack': 62, 'make_coffee_or_tea_/_use_a_coffee_machine': 63, 'fill_a_pot_/_bottle_/_container_with_water': 64, 'drink_beverage': 65, 'cut_open_a_package_(e.g._with_scissors)': 66, 'serve_food_onto_a_plate': 67, 'wash_dishes_/_utensils_/_bakeware_etc.': 68, 'prepare_or_apply_cement_/_concrete_/_mortar': 69, 'move_/_shift_around_construction_material': 70, 'rinse_/_drain_other_food_item_in_sieve_/_colander': 71, 'clean_/_wipe_/_oil_metallic_item': 72, 'pack_other_items_into_bags_/_boxes': 73, 'peel_a_fruit_or_vegetable': 74, '"cut_/_chop_/_slice_a_vegetable,_fruit,_or_meat"': 75, 'cut_/_trim_grass_with_a_lawnmower': 76, '"try-out_/_wear_clothing_items_(e.g._shirt,_jeans,_sweater)"': 77, 'move_/_shift_/_arrange_small_tools': 78, 'play_a_video_game': 79, 'do_some_exercise': 80, 'put_away_(or_take_out)_dishes_/_utensils_in_storage': 81, 'chop_/_cut_wood_pieces_using_tool': 82, 'look_at_clothes_in_the_mirror': 83, 'fix_/_remove_/_replace_a_tire_or_wheel': 84, 'remove_weeds_from_ground': 85, 'harvest_vegetables_/_fruits_/_crops_from_plants_on_the_ground': 86, 'fix_pipe_/_plumbing': 87, 'smooth_wood_using_sandpaper_/_sander_/_tool': 88, 'load_/_unload_a_washing_machine_or_dryer': 89, 'cut_tree_branch': 90, 'collect_/_rake_dry_leaves_on_ground': 91, 'cut_/_trim_grass_with_other_tools': 92, 'smoke_cigar_/_cigarette_/_vape': 93, 'iron_clothes_or_sheets': 94, 'wash_vegetable_/_fruit_/_food_item': 95, 'taste_food_while_cooking': 96, 'compare_two_clothing_items': 97, 'dig_or_till_the_soil_by_hand': 98, 'drive_a_vehicle': 99, 'trim_hedges_or_branches': 100, 'interact_or_play_with_pet_/_animal': 101, '"try-out_/_wear_accessories_(e.g._tie,_belt,_scarf)"': 102, 'tie_up_branches_/_plants_with_string': 103, 'arrange_pillows_on_couch_/_chair': 104, '"make_the_bed_/_arrange_pillows,_sheets_etc._on_bed"': 105, 'put_food_into_the_oven_to_bake': 106, 'fix_bonnet_/_engine_of_car': 107, 'write_notes_in_a_paper_/_book': 108, 'hang_clothes_to_dry': 109}
[INIT] Loaded annotations from validation subset.
        Number of ground truth instances: 4296
        Number of predictions: 104200
        Fixed threshold for tiou score: [0.1, 0.2, 0.3, 0.4, 0.5]
Warning: No predictions of label 'remove_food_from_the_oven' were provided.
Warning: No predictions of label 'put_food_into_the_oven_to_bake' were provided.
[RESULTS] Performance on Ego4D detection task.
Average-mAP: 0.05360927271911966
mAPs are [0.07966768 0.06482197 0.05193738 0.04003949 0.03157985]
mAP at tIoU 0.1 is 0.07966767691516692
mAP at tIoU 0.2 is 0.06482196566549295
mAP at tIoU 0.3 is 0.05193738098020502
mAP at tIoU 0.4 is 0.04003949305069336
mAP at tIoU 0.5 is 0.03157984698404002
Detection evaluation finishes!

---------------------------------------------------------------------------------------------
3. Retrieval evaluation starts!
---------------------------------------------------------------------------------------------
a. Generate retrieval!
b. Evaluate the retrieval results!
[INIT] Loaded annotations from validation subset.
        Number of ground truth instances: 521
        Number of predictions: 521
        Fixed threshold for tiou score: [0.1, 0.2, 0.3, 0.4, 0.5]
Rank 1x @ tIoU 0.3 is 0.32146182495344505
Rank 2x @ tIoU 0.3 is 0.449487895716946
Rank 3x @ tIoU 0.3 is 0.5139664804469274
Rank 4x @ tIoU 0.3 is 0.5616852886405959
Rank 5x @ tIoU 0.3 is 0.5926443202979516
Rank 1x @ tIoU 0.5 is 0.239292364990689
Rank 2x @ tIoU 0.5 is 0.3466014897579143
Rank 3x @ tIoU 0.5 is 0.4029329608938548
Rank 4x @ tIoU 0.5 is 0.44972067039106145
Rank 5x @ tIoU 0.5 is 0.4788175046554935
Rank 1x @ tIoU 0.7 is 0.15409683426443202
Rank 2x @ tIoU 0.7 is 0.2164804469273743
Rank 3x @ tIoU 0.7 is 0.24581005586592178
Rank 4x @ tIoU 0.7 is 0.2723463687150838
Rank 5x @ tIoU 0.7 is 0.2886405959031657
[[0.32146182 0.4494879  0.51396648 0.56168529 0.59264432]
 [0.23929236 0.34660149 0.40293296 0.44972067 0.4788175 ]
 [0.15409683 0.21648045 0.24581006 0.27234637 0.2886406 ]]
Detection evaluation finishes!

```

Thanks,
Junwei

**Comment by JunweiLiang** [comment]
@ebyrne

**Comment by JunweiLiang** [comment]
After reading the paper [EgoVLP](https://arxiv.org/pdf/2206.01670.pdf), I think the code works fine as the reported validation average-mAP is 6.03 vs. 5.36 of my run.

**Comment by JonnyS1226** [comment]
> 
hello, sorry to bother you. I want to ask what's the gpu memory is needed for training?  When i attempt to train, i found that it needs huge gpu memory because of KNN. I wonder if there's some errors with my cuda version etc. or this experiment exactly needs large gpu memory.
